lfsapi/client.go

```diff
@@ -75,6 +77,41 @@ func (c *Client) Do(req *http.Request) (*http.Response, error) {
 	return res, c.handleResponse(res)
 }
 
+func (c *Client) setExtraHeaders(req *http.Request) http.Header {
+	var copy http.Header = make(http.Header)
+	for k, vs := range req.Header {
+		copy[k] = vs
+	}
```

Comment:
> Why bother making a copy? Can't you just modify the req headers in place?

Good question. I think it would use less memory to modify the headers in place (though not much). My thought with making the copy was to localize mutating the request to outside of this function so that this could be more easily tested/changed in the future. I'm not tied at all to this implementation, so if you feel strongly, I'm happy to change it.

I pushed https://github.com/git-lfs/git-lfs/pull/2159/commits/d4aec737b2e3a5c8f33517ec9f3a04137ab61a5e to clarify the name of this method, and https://github.com/git-lfs/git-lfs/pull/2159/commits/961456e88f831608b3c43342ea536d7ebd5e1e68 to remove the extraneous `var` decl. I prefer those for initializing values that are going to change immediately, but I don't mind either way.